change/@ni-nimble-components-c29f703d-f01e-45c4-942f-c887470025b8.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```

Comment:
> 🧪 Testing
Added/updated tests

You should also do some manual testing with these changes to make sure this approach doesn't have accessibility violations associated with it. Specifically, when I go to the "Icon Column" story in storybook, I see new accessibility violations associated with the icons & their underlying `svg` elements.

Reply:
I had tested before explicitly adding `role="img"` to the inner `svg`s, and there were no issues. Interestingly, even though the `svg` element seems to have a default role of `img`, it is only when we explicitly set `role="img"` do we get violations. This is kind of a can of worms, because now every icon in our components needs to set `alt` text to avoid these violations. Looks like I'm going to have to add some new tokens for our label providers.

Reply:
Should the expectation be that every icon has alt text no matter where it is used? For example, do we want a button that says "+  Add column" to bubble up in the accessibility tree alt text for the "+" icon within the button? Or, do you think this accessibility violation is an indication that marking our icons as an `img` is the wrong direction?

Reply:
Did some more research. I think it is incorrect to always set the role to `img`. In your "+ Add column" example, or for the down arrow of a nimble-select (which is not a separate button), I think the correct role is `presentation`/`none`, since the image is purely decorational. It also appears that there is a more specific/appropriate role to use when the icon has meaning: `graphics-symbol`. This role was introduced in ARIA 1.1, so we can also specify `img` as a fallback role. We want to sometimes set the role to `presentation` and sometimes `graphics-symbol`. We would rather not leave it up to the client to set `role` and `aria-label` on the host `nimble-icon-*` element, so I suggest we use the presence of an `alt` text value to determine which role to set on the `svg`.

Note also that Storybook's Accessibility tab was giving us false negatives before I made the change in this PR. The default role of `svg` is `graphics-document` (even though Chrome shows it as `img`) which also requires an accessible name.

Sources:
https://www.w3.org/TR/graphics-aria-1.0/#graphics-symbol
https://www.unimelb.edu.au/accessibility/techniques/accessible-svgs
https://www.w3.org/WAI/ARIA/apg/practices/hiding-semantics/

Reply:
It doesn't look like your implementation/proposal follows the best practices included in the [accessible-svgs](https://www.unimelb.edu.au/accessibility/techniques/accessible-svgs) page you linked. In the `Hiding SVGs` section, it says you shouldn't set the role to `presentation` -- instead, you should set `aria-hidden`. Is there a reason you're doing something different here?

Reply:
Good catch. I had missed that detail.

Reply:
Apparently, Chromatic's Accessibility checks don't like fallback roles, so now I'm just using `graphics-symbol` without a fallback role of `img`.

Reply:
A few thoughts:
1. Do we have any idea of how widely supported ARIA 1.1 is? That's when you said `graphics-symbol` was added, right? If we can't do the `role` fallback, we should make sure we're picking a `role` that's well supported.
2. Perhaps we should have a larger discussion on the direction this has taken before getting into the details of the implementation in this PR. I'm not aware of any other component where we change its `role` based on a seemingly unrelated attribute, so I think it would be good for us to make sure we're happy with that as a new pattern here.